controllers/vm_controller.go

```diff
@@ -0,0 +1,111 @@
+package controllers
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"k8s.io/apimachinery/pkg/api/errors"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+const (
+	vmControllerName = "vm-controller"
+	rhel6MetricName  = "kubevirt_vm_rhel6"
+)
+
+var (
+	VmRhel6 = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: rhel6MetricName,
+			Help: "Indication for a VirtualMachine that is based on rhel6 template",
+		},
+		[]string{
+			"namespace", "name",
+		},
+	)
+)
+
+// Annotation to generate RBAC roles to read virtualmachines
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachines,verbs=get;list;watch
+
+func CreateVmController(mgr ctrl.Manager) (*vmReconciler, error) {
+	return newVmReconciler(mgr)
+}
+
+func (r *vmReconciler) Name() string {
+	return vmControllerName
+}
+
+func (r *vmReconciler) Start(ctx context.Context, mgr ctrl.Manager) error {
+	return r.setupController(mgr)
+}
+
+func (r *vmReconciler) setupController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("vm-controller").
+		For(&kubevirtv1.VirtualMachine{}, builder.WithPredicates(predicate.NewPredicateFuncs(
+			func(object client.Object) bool {
+				return hasRhel6TemplateLabel(object)
+			}))).
+		Complete(r)
+}
+
+// vmReconciler watches the vms in the cluster
+type vmReconciler struct {
+	client client.Client
+	log    logr.Logger
+}
+
+func newVmReconciler(mgr ctrl.Manager) (*vmReconciler, error) {
+	logger := ctrl.Log.WithName("controllers").WithName("VirtualMachines")
+	reconciler := &vmReconciler{
+		client: mgr.GetClient(),
+		log:    logger,
+	}
+
+	return reconciler, nil
+}
+
+func (r *vmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	r.log.Info("Starting vm reconciliation...", "request", req.String())
```

Comment:
This log will be to verbose, because it will be printed for every VM creation, update and delete.
Can you change it to debug level using `.V(1)`? Like here: https://github.com/kubevirt/ssp-operator/blob/aabcf8d8d1c61ca04b97b4c830c2ff4c14f88458/controllers/ssp_controller.go#L212

Reply:
Totally agree, thanks